### PR TITLE
[1.19.x] Update to FG6 and Gradle 8, backport 1.20 MDK

### PR DESCRIPTION
**Pull request by PaintNinja**
Backport of #9750, #9596, #9592 to 1.19.4.

Also ports Forge 1.19.4 to FG6 and Gradle 8, with the accompanying MDK updates. While I was at it I backported most of the new MDK's features as well.

### CODE REVIEW

File: mdk/README.txt [hunk 1]
```diff
@@ -16,14 +16,14 @@ Step 1: Open your command-line and browse to the folder where you extracted the
 
 Step 2: You're left with a choice.
 If you prefer to use Eclipse:
-1. Run the following command: `gradlew genEclipseRuns` (`./gradlew genEclipseRuns` if you are on Mac/Linux)
 2. Open Eclipse, Import > Existing Gradle Project > Select Folder 
+1. Run the following command: `./gradlew genEclipseRuns`
    or run `gradlew eclipse` to generate the project.
```
Comment (Jonathing) [inline mdk/README.txt:21]: Does `gradlew eclipse` not work anymore? Or well, I guess it doesn't in FG6. I don't remember what its deal is in FG5.

**Review by Jonathing — APPROVED** [review]

File: mdk/README.txt (hunk 1 above)
Comment (PaintNinja) [reply mdk/README.txt:21]: Dunno, I don't use neither Eclipse or FG5 anymore :P

**Comment by PaintNinja** [comment]
Bad autoforge, bad!

**Comment by LexManos** [comment]
> Sidenote: We should consider updating to FG6 for next 1.19.4 RB

Why not do it now?

**Comment by Jonathing** [comment]
True! I've been using FG6 on my personal 1.19.4 workspace with no issues so far.

**Comment by PaintNinja** [comment]
Because I don't know the procedure for updating Gradle and FG on the existing Forge buildscript we've got, which is required due to the way the MDK's generated (can't simply bump FG and Gradle for the MDK on its own)

**Comment by LexManos** [comment]
`gradlew wrapper --wrapper-version <?>`  and then fix everything that explodes.
FG6 should run on everything that FG5 runs on, it was just a gradle backend bump IIRC.

**Pull request merged**